resource/interface/client/skillcheck.lua

```diff
@@ -0,0 +1,62 @@
+---@type promise?
+local skillcheck
+local gameCount = 0
+
+---@alias SkillCheckDifficulity 'easy' | 'medium' | 'hard' | { areaSize: number, speedMultiplier: number }
+
+---@type SkillCheckDifficulity | SkillCheckDifficulity[]
+local gameDifficulty
+
+---@type promise?
+local gamePromise
+
+local function resolveSkillCheck(success)
+    if skillcheck then
+        skillcheck:resolve(success)
+        skillcheck = nil
+        SetNuiFocus(false, false)
+    end
+end
+
+---@param difficulty SkillCheckDifficulity | SkillCheckDifficulity[]
+---@return boolean?
+function lib.skillCheck(difficulty)
+    if skillcheck then return end
+
+    skillcheck = promise:new()
+    gameDifficulty = difficulty
+    SetNuiFocus(true, false)
+
+    if type(difficulty) == 'table' then
+        gameCount = 0
+        for i = 1, #difficulty do
+            gamePromise = promise:new()
+            SendNUIMessage({
+                action = 'startSkillCheck',
+                data = difficulty[i]
+            })
+            if not Citizen.Await(gamePromise) then
+                gamePromise = nil
+                break
+            end
+        end
+    else
+        SendNUIMessage({
+            action = 'startSkillCheck',
+            data = difficulty
+        })
+    end
```

Comment:
Would be best to handle this in the ui; 'startSkillCheck' should await all promises to resolve before triggering 'skillCheckOver'.

Reply:
Also just noticed difficulty can be a string, map, or array (containing strings or maps), so sending a custom difficulty wouldn't work 🤔
```lua
lib.skillCheck({areaSize = 250, speedMultiplier = 2})
```
This would fix it, but should still be done in nui anyway.
```lua
if type(difficulty) == 'table' and table.type(difficulty) == 'array' then
```

Reply:
It will all be solved once I move the logic into UI